tex/gregoriotex.lua

```diff
@@ -1237,101 +1174,80 @@ local function locate_file(filename)
   return result
 end
 
-local function include_score(input_file, force_gabccompile, allow_deprecated)
-  if string.match(input_file, "[#%%]") then
+local function include_score(gabc_file, force_gabccompile, allow_deprecated)
+  local sep = ""
+  local onwindows = (os.type == "windows" or
+                     string.find(os.getenv("PATH"),";",1,true))
+  if onwindows then
+    sep = "\\"
+  else
+    sep = "/"
+  end
+    
+  if string.match(gabc_file, "[#%%]") then
     err("GABC filename contains invalid character(s): # %%\n"
-        .."Rename the file and retry: %s", input_file)
+        .."Rename the file and retry: %s", gabc_file)
   end
-  local has_extention = false
-  local file_dir,input_name
+  local gabc_dir, base
   local extensions = {['gabc']=true, ['gtex']=true, ['tex']=true}
-  if extensions[string.match(input_file, "([^%.\\/]*)$")] then
-    has_extention = true
-  end
-  if has_extention then
-    file_dir,input_name = string.match(input_file, "(.-)([^\\/]-)%.?[^%.\\/]*$")
+  if extensions[string.match(gabc_file, "([^%.\\/]*)$")] then
+    gabc_dir, base = string.match(gabc_file, "(.-)([^\\/]-)%.?[^%.\\/]*$")
   else
-    file_dir,input_name = string.match(input_file, "(.-)([^\\/]*)$")
-  end
-
-  local cleaned_filename = input_name:gsub("[%s%+%&%*%?$@:;!\"\'`]", "-")
-  local gabc_filename = string.format("%s%s.gabc", file_dir, input_name)
-  local gabc_file = locate_file(gabc_filename)
-  local gtex_filename = string.format("%s%s-%s.gtex", file_dir, cleaned_filename,
-      internalversion:gsub("%.", "_"))
-  local gtex_file = locate_file(gtex_filename)
-  local glog_file = string.format("%s%s-%s.glog", file_dir, cleaned_filename,
-      internalversion:gsub("%.", "_"))
-  if not gtex_file then
-    clean_old_gtex_files(file_dir..cleaned_filename)
```

Comment:
This was cleaning gtex files in the wrong place, and I just deleted it. But it can be put back and made to clean gtex files inside tmp-gre.

Reply:
What do you mean by "wrong place"?  Before the tmp-gre system, gtex files were placed in the same directory as the gabc files and all this was meant to clean-up those gtex files.  We probably should still clean those files up, or at least indicate to the users (perhaps in UPGRADE.md) that they should be deleting those files themselves.

Reply:
By "wrong place" I mean the "old place". I can restore the function and the call to it, and are you saying you want to clean them up in both the old place and the new place?

(My opinion is that programs should avoid removing files without asking the user.)

Reply:
We decided to remove these files to prevent excessive clutter of gtex files made for old versions of gregorio. Since these gtex files were created by gregorio we felt that gregorio should be allowed to remove them without user input.

Reply:
I agree with @eschwab.  We create these files so we should be allowed to delete them.